chia-bls/src/gtelement.rs

```diff
@@ -0,0 +1,159 @@
+use blst::*;
+use chia_traits::chia_error::Result;
```

Comment:
Why are we reusing this error type for other crates? Would be better for `chia-bls` to have its own concrete error enum with common variants (can use `thiserror` or `impl std::error::Error` to make it a proper error type) instead of a generic `Custom` String variant.

Reply:
no good reason, just laziness. I'm not familiar with `thiserror`. Do you think we should stick to that in the chia repos?

Reply:
It makes it easy to derive From for wrapped types, set how each variant is Display'ed, and derive the Error trait

Reply:
I believe this has been addressed outside of this PR now (this PR has been rebased on top of it)